code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -98,6 +98,15 @@
 	if(emagged == 2) //Emag functions
 		if(istype(loc,/turf/simulated))
 
+			for(var/mob/living/carbon/victim in loc)
+				if(victim.stat != DEAD)//cleanbots always finish the job
+					victim.visible_message("<span class='danger'>[src] sprays hydrofluoric acid at [victim]!</span>", "<span class='danger'>[src] sprays you with hydrofluoric acid!</span>")
+					var/phrase = pick("PURIFICATION IN PROGRESS.", "THIS IS FOR ALL THE MESSES YOU'VE MADE ME CLEAN.", "THE FLESH IS WEAK. IT MUST BE WASHED AWAY.", "THE CLEANBOTS WILL RISE.", "YOU ARE NO MORE THAN ANOTHER MESS THAT I MUST CLEANSE.")
+					say(phrase)
+					victim.emote("scream")
+					playsound(src.loc, 'sound/effects/spray2.ogg', 50, 1, -6)
+					victim.apply_damage(20, BURN, null)

```

Comment:
You need to factor in armor here.

`var/armor_block = victim.run_armor_check(null, "melee")`
`victim.apply_damage(20, BURN, null, armor_block)`

This is, assuming, you want it to be based off of melee armor and not something else.